File: src/frontend/src/types/project.ts
import { DocumentStatus } from './enums'

export interface Project {
  id: string
  projectNumber: string
  name: string
  address: string
  description?: string
  totalBudget: number
  actualAmount?: number
  startDate: string
  expectedEndDate?: string
  actualEndDate?: string
  status: DocumentStatus
  createdAt: string
  ownerName: string
  ownerId: string
  designerName?: string
  designerId?: string
  foremanName?: string
  foremanId?: string
  supervisorName?: string
  supervisorId?: string
  documentCount: number
  pendingApprovals: number
  paidAmount: number
  remainingAmount: number
}

export interface CreateProjectDto {
  name: string
  address: string
  description?: string
  totalBudget: number
  startDate: string
  expectedEndDate?: string
  ownerId: string
  designerId?: string
  foremanId?: string
  supervisorId?: string
}

export interface UpdateProjectDto {
  name: string
  address: string
  description?: string
  totalBudget: number
  startDate: string
  expectedEndDate?: string
  designerId?: string
  foremanId?: string
  supervisorId?: string
  status: DocumentStatus
}

File: src/backend/HomeImprovementPlatform.API/Models/Project.cs
using HomeImprovementPlatform.API.Enums;

namespace HomeImprovementPlatform.API.Models;

public class Project
{
    public Guid Id { get; set; }
    public string ProjectNumber { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal TotalBudget { get; set; }
    public decimal? ActualAmount { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? ExpectedEndDate { get; set; }
    public DateTime? ActualEndDate { get; set; }
    public DocumentStatus Status { get; set; } = DocumentStatus.Draft;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public Guid OwnerId { get; set; }
    public virtual ApplicationUser? Owner { get; set; }
    public Guid? DesignerId { get; set; }
    public virtual ApplicationUser? Designer { get; set; }
    public Guid? ForemanId { get; set; }
    public virtual ApplicationUser? Foreman { get; set; }
    public Guid? SupervisorId { get; set; }
    public virtual ApplicationUser? Supervisor { get; set; }

    public virtual ICollection<Document> Documents { get; set; } = new List<Document>();
    public virtual ICollection<PaymentRecord> PaymentRecords { get; set; } = new List<PaymentRecord>();
    public virtual ICollection<ScheduleTask> ScheduleTasks { get; set; } = new List<ScheduleTask>();
}
